sedaro/src/sedaro/branches/scenario_branch/sim_client.py

```diff
@@ -253,62 +253,60 @@ def __fetch(
             url += f'&continuationToken={continuationToken}'
         url += '&encoding=msgpack'
 
-        self.dmlog(download_manager, f"Calling URL: {url}")
-        response = fast_fetcher.get(url)
-        self.dmlog(download_manager, f"Got response from {url} -- status: {response.status}. Parsing response...")
-        _response = None
-        has_nonempty_ctoken = False
-        try:
-            _response = response.parse()
-            self.dmlog(download_manager, f"Parsed response from {url}")
-            if response.status != 200:
-                raise Exception()
-            else:
-                download_manager.ingest(_response['series'])
-                download_manager.add_metadata(_response['meta'])
-                if 'continuationToken' in _response['meta'] and _response['meta']['continuationToken'] is not None:
-                    has_nonempty_ctoken = True
-                    ctoken = _response['meta']['continuationToken']
-                if 'stats' in _response:
-                    download_manager.update_stats(_response['stats'])
-                if 'derived' in _response:
-                    if 'series' in _response['derived']:
-                        download_manager.ingest_derived(_response['derived']['series'])
-                    if 'static' in _response['derived']:
-                        download_manager.update_static_data(_response['derived']['static'])
-        except Exception as e:
-            reason = _response['error']['message'] if _response and 'error' in _response else 'An unknown error occurred.'
-            raise SedaroApiException(status=response.status, reason=reason)
-        if has_nonempty_ctoken:  # need to fetch more pages
-            while has_nonempty_ctoken:
-                # fetch page
-                request_url = f'/data/{id}?&continuationToken={ctoken}'
+        def get_and_parse_page_with_retry(
+            fetcher: FastFetcher,
+            request_url: str,
+            download_manager: DownloadWorker,
+            max_retries: int = 5,
+        ):
+            for attempt in range(max_retries):
                 self.dmlog(download_manager, f"Calling URL: {request_url}")
-                page = fast_fetcher.get(request_url)
+                raw_response = fetcher.get(request_url)
+                if raw_response.status == 500:
+                    # retry on 500 error
+                    self.dmlog(download_manager, f"Received 500 error from {request_url}. Retrying...")
+                    continue
```

Comment:
Let me know if this wouldn't make sense here, but could it be worth adding a short sleep in here? If it made sense, the time could also be increased a little bit each time

time.sleep(1.5 ** attempt)